From: Alexander Orlovskiy <nordicdyno@gmail.com>
Date: Wed, 15 Aug 2012 14:48:31 +0400
Subject: [PATCH] Workaround for passing tests in Perl verisons >= 5.16

  - remove code that never works, but caused problem with warning message
    "Can't modify non-lvalue subroutine" in tests
---
 lib/SOAP/WSDL.pm | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/lib/SOAP/WSDL.pm b/lib/SOAP/WSDL.pm
index 6a753ea..b11b5f5 100644
--- a/lib/SOAP/WSDL.pm
+++ b/lib/SOAP/WSDL.pm
@@ -164,18 +164,14 @@ sub wsdlinit {
 sub _wsdl_get_service :PRIVATE {
     my $ident = ident shift;
     my $wsdl = $definitions_of{ $ident };
-    return $service_of{ $ident } = $servicename_of{ $ident }
-        ? $wsdl->find_service( $wsdl->get_targetNamespace() , $servicename_of{ $ident } )
-        : $service_of{ $ident } = $wsdl->get_service()->[ 0 ];
+    return $service_of{ $ident } = $wsdl->get_service()->[ 0 ];
 } ## end sub _wsdl_get_service
 
 sub _wsdl_get_port :PRIVATE  {
     my $ident = ident shift;
     my $wsdl = $definitions_of{ $ident };
     my $ns   = $wsdl->get_targetNamespace();
-    return $port_of{ $ident } = $portname_of{ $ident }
-        ? $service_of{ $ident }->get_port( $ns, $portname_of{ $ident } )
-        : $port_of{ $ident } = $service_of{ $ident }->get_port()->[ 0 ];
+    return $port_of{ $ident } = $service_of{ $ident }->get_port()->[ 0 ];
 }
 
 sub _wsdl_get_binding :PRIVATE {
-- 
1.7.11.4

